internal/extract.go

```diff
@@ -218,34 +171,30 @@ func tryExtractFiles(files []ReaderMaker,
 			tracelog.ErrorLogger.Println(err)
 			return files //Should never happen, but if we are asked to cancel - consider all files unfinished
 		}
-		err = writingSemaphore.Acquire(downloadingContext, 1)
```

Comment:
Why did you remove the `writingSemaphore` logic?

Reply:
There is now only 1 goroutine which decompresses & writes, instead of sending decompresssed bytes to pipe which gets written

Reply:
> There is now only 1 goroutine which decompresses & writes, instead of sending decompresssed bytes to pipe which gets written
> 
> These are all still capped by a `semaphore.NewWeighted(int64(downloadingConcurrency))`

Oh, looks like I've missed it...